editor/editor_sdk_manager.cpp

```diff
@@ -0,0 +1,378 @@
+}
+
+void EditorSDKManager::_update_packages_thread(void *s) {
+	EditorSDKManager *self = (EditorSDKManager *)s;
+	self->_update_packages();
+}
+
+void EditorSDKManager::_update_packages() {
+	List<String> args;
+	args.push_back("--list");
+
+	String output;
+	int result = _run_sdkmanager(args, &output);
+	if (result != OK) {
+		_is_working = false;
+		return;
+	}
+
+	List<String> installed;
+
+	int from = output.find("Installed packages:");
+	int limit = output.find("Available Packages:");
+
+	// Get the paths of the installed packages.
+	if (from != -1) {
+		from = output.find("\n", output.find("-------", from)) + 1;
+
+		while (true) {
+			int to = output.find("|", from) ;
+			if (to > limit) {
+				break;
+			}
+
+			installed.push_back(output.substr(from, to - from).strip_edges());
+
+			from = output.find("\n", from) + 1;
+			if (from >= limit) {
+				break;
+			}
+		}
+	}
+
+	from = output.find("\n", output.find("-------", limit)) + 1;
+	limit = output.length() - 1;
+
+	// Extract the info of all packages available.
+	packages_available.clear();
+	while (true) {
+		int newline = output.find("\n", from);
+		if (newline == -1) {
+			break;
+		}
+
+		String line = output.substr(from, newline - from).strip_edges();
+		if (!line.empty()) {
+			Vector<String> strings = output.substr(from, newline - from).split("|", false);
+			ERR_BREAK(strings.size() < 2);
+
+			Vector<String> data;
+			data.push_back(strings[2].strip_edges());
+			data.push_back(strings[1].strip_edges());
+			packages_available[strings[0].strip_edges()] = data;
+		}
+
+		from = newline + 1;
+	}
+
+	if (!installed.empty()) {
+		//Place the installed ones in their own container.
+		for (List<String>::Element *E = installed.front(); E; E = E->next()) {
+			if (packages_available.has(E->get())) {
+				packages_installed[E->get()] = packages_available[E->get()];
+				packages_available.erase(E->get());
+			} else {
+				Vector<String> data;
+				data.push_back(E->get());
+				data.resize(2);
+				packages_installed[E->get()] = data;
+			}
+		}
+	}
+
+	_is_working = false;
+}
+
+void EditorSDKManager::_apply_changes_thread(void *s) {
+	EditorSDKManager *self = (EditorSDKManager *)s;
+	self->_apply_changes();
+}
+
+void EditorSDKManager::_apply_changes() {
+	if (!to_remove.empty()) {
+		List<String> args;
+		args.push_back("--uninstall");
+
+		for (List<String>::Element *E = to_remove.front(); E; E = E->next()) {
+			args.push_back("\"" + E->get() + "\"");
+		}
+
+		String output;
+		int result = _run_sdkmanager(args, &output);
+		if (result != OK) {
+			_is_working = false;
+			return;
+		}
+
+		for (List<String>::Element *E = to_remove.front(); E; E = E->next()) {
+			packages_available[E->get()] = packages_installed[E->get()];
+			packages_installed.erase(E->get());
+		}
+		to_remove.clear();
+	}
+
+	if (!to_install.empty()) {
+		List<String> args;
+		for (List<String>::Element *E = to_install.front(); E; E = E->next()) {
+			args.push_back("\"" + E->get() + "\"");
```

Comment:
Remove the quotes here, OS::execute() already adds quotes to all parameters. Just leave as `args.push_back(E->get());`